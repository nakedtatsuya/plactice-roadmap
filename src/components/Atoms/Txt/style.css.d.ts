export const xxs: string;
export const xs: string;
export const s: string;
export const m: string;
export const l: string;
export const xl: string;
export const xxl: string;
export const xxxl: string;
export const xxxxl: string;
export const base: string;
export const white: string;
export const success: string;
export const info: string;
export const primary: string;
export const warning: string;
export const danger: string;
