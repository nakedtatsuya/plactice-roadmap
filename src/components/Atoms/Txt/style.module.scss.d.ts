declare const styles: {
  readonly "xxs": string;
  readonly "xs": string;
  readonly "s": string;
  readonly "m": string;
  readonly "l": string;
  readonly "xl": string;
  readonly "xxl": string;
  readonly "xxxl": string;
  readonly "xxxxl": string;
  readonly "base": string;
  readonly "white": string;
  readonly "success": string;
  readonly "info": string;
  readonly "primary": string;
  readonly "warning": string;
  readonly "danger": string;
};
export = styles;

