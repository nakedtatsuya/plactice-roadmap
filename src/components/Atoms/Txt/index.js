import React from "react";
import styles from "./styles.module.scss";
import PropTypes from "prop-types";

const txtFactory = role => ({ tag: Tag, fontSize, className, ...props }) => (
  <Tag {...props} className={[styles[role], styles[fontSize], className].join(" ")} />
);

const Txt = txtFactory("base");

Txt.propTypes = {
  children: PropTypes.node.isRequired,
  tag: PropTypes.string,
  fontSize: PropTypes.string,
};

Txt.defaultProps = {
  tag: "p",
  fontSize: "m",
};

export default Txt;
