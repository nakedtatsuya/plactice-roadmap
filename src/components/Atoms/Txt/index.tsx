import React from 'react';
import style from './style.module.scss';
import { TxtRole, FontSize, TagType } from '../../../constants';
import { ITxt } from '../../../types/txt';

const txtFactory = (role: TxtRole): React.FC<ITxt> => ({
  tag: Tag = TagType.p,
  fontSize = FontSize.m,
  className,
  ...props
}) => {
  return (
    <Tag
      {...props}
      className={[style[role], style[fontSize], className].join(' ')}
    />
  );
};

const Txt = txtFactory(TxtRole.base);
export const WhiteTxt = txtFactory(TxtRole.white);
export const InfoTxt = txtFactory(TxtRole.info);
export const SuccessTxt = txtFactory(TxtRole.success);
export const WarningTxt = txtFactory(TxtRole.warning);
export const DangerTxt = txtFactory(TxtRole.danger);
export const PrimaryTxt = txtFactory(TxtRole.primary);

export default Txt;
