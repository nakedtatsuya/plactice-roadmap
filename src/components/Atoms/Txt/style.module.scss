@import url("../../properties.css");

.xxs {
    font-size: var(--font-size-xxs);
}

.xs {
    font-size: var(--font-size-xs);
}

.s {
    font-size: var(--font-size-s);

}

.m {
    font-size: var(--font-size-m);
}

.l {
    font-size: var(--font-size-l);
}

.xl {
    font-size: var(--font-size-xl);
}

.xxl {
    font-size: var(--font-size-xxl);
}

.xxxl {
    font-size: var(--font-size-xxxl);
}

.xxxxl {
    font-size: var(--font-size-xxxxl);
}

.base {
    color: var(--color-text);
}

.white {
    color: var(--color-white);
}

.success {
    color: var(--color-success);
}

.info {
    color: var(--color-info);
}

.primary {
    color: var(--color-primary);
}

.warning {
    color: var(--color-warning);
}

.danger {
    color: var(--color-danger);
}
