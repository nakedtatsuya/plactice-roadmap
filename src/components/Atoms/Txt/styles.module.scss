@import '../../properties';

.txt {
  line-height: var(--line-height-m);
}

.visualLevel1 {
  font-size: var(--font-size-m);

  @include mediaQuery(md) {
    font-size: var(--font-size-s);
  }
}

.visualLevel2 {
  font-size: var(--font-size-s);

  @include mediaQuery(md) {
    font-size: var(--font-size-xs);
  }
}

.visualLevel3 {
  font-size: var(--font-size-xs);

  @include mediaQuery(md) {
    font-size: var(--font-size-xxs);
  }
}

.visualLevel4 {
  font-size: var(--font-size-xxs);
}

.base {
  color: var(--color-text);
}

.info {
  color: var(--color-text-info);
}

.disable {
  color: var(--color-text-disable);
}
