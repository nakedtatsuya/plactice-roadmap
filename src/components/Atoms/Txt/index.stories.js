import React from 'react'
import {storiesOf} from '@storybook/react'
import Txt, {InfoTxt} from './index'

storiesOf(`Atoms/Txt`, module)
  .add(`Text - XXS`, () => <Txt fontSize={'xxs'}>Show text</Txt>)
  .add(`Text - XS`, () => <Txt fontSize={'xs'}>Show text</Txt>)
  .add(`Text - S`, () => <Txt fontSize={'s'}>Show text</Txt>)
  .add(`Text - M`, () => <Txt>Show text</Txt>)
  .add(`Text - L`, () => <Txt fontSize={'l'}>Show text</Txt>)
  .add(`Text - XL`, () => <Txt fontSize={'xl'}>Show text</Txt>)
  .add(`Text - XXL`, () => <Txt fontSize={'xxl'}>Show text</Txt>)
  .add(`Text - XXXL`, () => <Txt fontSize={'xxxl'}>Show text</Txt>)
  .add(`Info Text`, () => <InfoTxt>Show text</InfoTxt>)
