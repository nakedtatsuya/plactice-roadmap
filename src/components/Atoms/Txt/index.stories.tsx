import {storiesOf} from "@storybook/react";
import Txt, {InfoTxt, DangerTxt, SuccessTxt, WarningTxt, PrimaryTxt, WhiteTxt} from "./index";
import {FontSize} from "../../../constants";
import React from "react";

storiesOf(`Atoms/Txt`, module)
.add(`テキスト - XXS`, () => (
		<Txt fontSize={FontSize.xxs}>テキストを表示</Txt>
))
.add(`テキスト - XS`, () => (
		<Txt fontSize={FontSize.xs}>テキストを表示</Txt>
))
.add(`テキスト - S`, () => (
		<Txt fontSize={FontSize.s}>テキストを表示</Txt>
))
.add(`テキスト - M`, () => (
		<Txt>テキストを表示</Txt>
))
.add(`テキスト - L`, () => (
		<Txt fontSize={FontSize.l}>テキストを表示</Txt>
))
.add(`テキスト - XL`, () => (
		<Txt fontSize={FontSize.xl}>テキストを表示</Txt>
))
.add(`テキスト - XXL`, () => (
		<Txt fontSize={FontSize.xxl}>テキストを表示</Txt>
))
.add(`テキスト - XXXL`, () => (
		<Txt fontSize={FontSize.xxxl}>テキストを表示</Txt>
))
.add(`テキスト - XXXXL`, () => (
		<Txt fontSize={FontSize.xxxxl}>テキストを表示</Txt>
))
.add(`情報テキスト - S`, () => (
		<InfoTxt fontSize={FontSize.s}>情報テキストを表示</InfoTxt>
))
.add(`情報テキスト - M`, () => (
		<InfoTxt>情報テキストを表示</InfoTxt>
))
.add(`情報テキスト - L`, () => (
		<InfoTxt fontSize={FontSize.l}>情報テキストを表示</InfoTxt>
))
.add(`サクセステキスト - S`, () => (
		<SuccessTxt fontSize={FontSize.s}>サクセステキストを表示</SuccessTxt>
))
.add(`サクセステキスト - M`, () => (
		<SuccessTxt>サクセステキストを表示</SuccessTxt>
))
.add(`サクセステキスト - L`, () => (
		<SuccessTxt fontSize={FontSize.l}>サクセステキストを表示</SuccessTxt>
))
.add(`危険テキスト - S`, () => (
		<DangerTxt fontSize={FontSize.s}>危険テキストを表示</DangerTxt>
))
.add(`危険テキスト - M`, () => (
		<DangerTxt>危険テキストを表示</DangerTxt>
))
.add(`危険テキスト - L`, () => (
		<DangerTxt fontSize={FontSize.l}>危険テキストを表示</DangerTxt>
))
.add(`警告テキスト - S`, () => (
		<WarningTxt fontSize={FontSize.s}>警告テキストを表示</WarningTxt>
))
.add(`警告テキスト - M`, () => (
		<WarningTxt>警告テキストを表示</WarningTxt>
))
.add(`警告テキスト - L`, () => (
		<WarningTxt fontSize={FontSize.l}>警告テキストを表示</WarningTxt>
))
.add(`優先テキスト - S`, () => (
		<PrimaryTxt fontSize={FontSize.s}>優先テキストを表示</PrimaryTxt>
))
.add(`優先テキスト - M`, () => (
		<PrimaryTxt>優先テキストを表示</PrimaryTxt>
))
.add(`優先テキスト - L`, () => (
		<PrimaryTxt fontSize={FontSize.l}>優先テキストを表示</PrimaryTxt>
))
.add(`ホワイトテキスト - S`, () => (
		<WhiteTxt fontSize={FontSize.s}>ホワイトテキストを表示</WhiteTxt>
))
.add(`ホワイトテキスト - M`, () => (
		<WhiteTxt>ホワイトテキストを表示</WhiteTxt>
))
.add(`ホワイトテキスト - L`, () => (
		<WhiteTxt fontSize={FontSize.l}>ホワイトテキストを表示</WhiteTxt>
))

