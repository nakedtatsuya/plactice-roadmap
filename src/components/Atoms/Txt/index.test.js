import React from "react";
import renderer from "react-test-renderer";
import Txt from "./index";

describe("Txt", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<Txt>Txt</Txt>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
