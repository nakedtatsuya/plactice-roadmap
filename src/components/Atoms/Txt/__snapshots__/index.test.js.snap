// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txt Renders correctly 1`] = `
<p
  className="txt base visualLevel1 "
>
  Txt
</p>
`;
